### PR TITLE
🏗️  Ignore .babel-cache directory

### PR DESCRIPTION
**Pull request by jridgewell**
Also ensure prettier respects its `.prettierignore` file, which for some reason it ignores when using the API vs the CLI.

### CODE REVIEW

File: build-system/tasks/prettify.js [hunk 1]
```diff
@@ -46,7 +47,10 @@ const tempDir = tempy.directory();
  * Checks files for formatting (and optionally fixes them) with Prettier.
  */
 async function prettify() {
-  const filesToCheck = getFilesToCheck(prettifyGlobs, {dot: true});
+  const ignored = ignore().add(fs.readFileSync('.prettierignore', 'utf8'));
+  const filesToCheck = ignored.filter(
+    getFilesToCheck(prettifyGlobs, {dot: true})
```
Comment (rsimha) [inline build-system/tasks/prettify.js:52]: `getFilesToCheck` calls [`globby.sync`](https://github.com/sindresorhus/globby#options), which uses [`fast-glob`](https://github.com/mrmlnc/fast-glob#ignore) under the covers. It should be sufficient to add an `ignore` field to the options object and pass in the list of lines in `.prettierignore`.

An alternative would be to add a new `ignoreFile` param to `getFilesToCheck` and have it read the file and construct the list of globs to ignore and pass them in to `globby.sync({ignore: ...})`.

Comment (jridgewell) [reply build-system/tasks/prettify.js:52]: I didn't want to add ignore file parsing to our code (comments, rules, etc). `ignore` is what both [prettier](https://github.com/prettier/prettier/blob/a17d3fb21f3dc45774f41a313d384ea156ae5b92/package.json#L54) and [eslint](https://github.com/eslint/eslint/blob/8984c91372e64d1e8dd2ce21b87b80977d57bff9/package.json#L67) use internally to create their ignore files, so it's what I went with.

Comment (rsimha) [reply build-system/tasks/prettify.js:52]: In that case, WDYT about the alternative suggested above? (Add an optional param called `ignoreFile` to `getFilesToCheck()` and if present, ignore the files listed in it.)

The goal is to be able to reuse this logic in other tasks that have their own ignore files. (e.g. lint can use this with `.eslintignore`)

**Review by rsimha — APPROVED** [review]

File: build-system/tasks/prettify.js (hunk 1 above)
Comment (jridgewell) [reply build-system/tasks/prettify.js:52]: Yah, I can relocate, but it's only prettier that needs it. (Prettier ignores the `.prettierignore` when using the API, and only uses it when using the CLI). Eslint and friends do the right thing. I'll add a comment explaining.

File: build-system/tasks/prettify.js [hunk 2]
```diff
@@ -46,7 +46,10 @@ const tempDir = tempy.directory();
  * Checks files for formatting (and optionally fixes them) with Prettier.
  */
 async function prettify() {
-  const filesToCheck = getFilesToCheck(prettifyGlobs, {dot: true});
+  // Prettier ignores the `.prettierignore` file when using the API like we
+  // are. So we need to filter our files down before feeding them to the API.
+  const ignore = fs.readFileSync('.prettierignore', 'utf8');
+  const filesToCheck = getFilesToCheck(prettifyGlobs, {dot: true, ignore});
```
Comment (rsimha) [inline build-system/tasks/prettify.js:52]: I think this should be passed in separately via a new param called `ignoreFile`. Directly adding an `ignore` field to `options` will pass it in to `globby.sync`, and I'm not sure the formats match. (This is a single string containing a block of files, which I'm not sure `globby.sync` knows to parse.)

Comment (jridgewell) [reply build-system/tasks/prettify.js:52]: Ahh, fixing.

**Pull request merged**